crates/ra_assists/src/handlers/replace_qualified_name_with_use.rs

```diff
@@ -12,10 +12,10 @@ use crate::{
     AssistId,
 };
 
-/// This function produces sequence of text edits into edit
-/// to import the target path in the most appropriate scope given
-/// the cursor position
-pub fn auto_import_text_edit(
+/// Creates and inserts a use statement for the given path to import.
+/// The use statement is inserted in the scope most appropriate to the
+/// the cursor position given, additionally merged with the existing use imports.
+pub fn insert_use_statement(
```

Comment:
FWIW, I feel it might makes sense to move this helper to the newly-introduced `utils` module. Ive tried to do this, but punted on this because it includes a lot of code

Reply:
Having exactly the same thoughts, had tried to to the same also, but everything is so coupled here, so no luck here yet.